src/components/VotingButton/VotingButton.component.tsx

```diff
@@ -0,0 +1,88 @@
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CircularProgress from '@mui/material/CircularProgress';
+import React, { useState } from 'react';
+import pendingIcon from '../../assets/icons/loading.svg';
+import { VoteType } from '../../constants';
+import { AppDialog } from '../AppDialog/AppDialog.component';
+import { VotingButtonProps } from './VotingButton.types';
+
+export const VotingButton = ({
```

Comment:
This component became a little bit too complicated. I think clearer solution would be to have a separated component only for the dialog and use it in both original `VoteForButton` and `VoteAgainstButton` components.